integration/management-api-tests/src/test/java/org/wso2/micro/integrator/api/ConfigsResourceTestCase.java

```diff
@@ -0,0 +1,290 @@
+     * config retrieves the proper status which is 'true'. It will check whether a PUT request
+     * will override the setting from the system property. ( Should not be able to override ).
+     * This is confirmed by reading the correlation log file again.
+     *
+     * @throws IOException                  the io exception
+     * @throws AutomationFrameworkException the automation framework exception
+     */
+    @Test(groups = {"wso2.esb"}, dependsOnMethods = {"testUpdateCorrelationConfigsDisable"},
+            description = "Test get configs resource for correlation with system parameter")
+    public void testRetrieveCorrelationConfigsWithSystemParameter() throws IOException, AutomationFrameworkException {
+
+        stopServer();
+        startNewServer(true);
+        sendGetRequest(offset, true);
+        sendPutRequest(offset, false);
+
+        String apiLogFilePath = carbonHome + File.separator + "repository"
+                + File.separator + "logs" + File.separator + "correlation.log";
+        BufferedReader bufferedReader = new BufferedReader(new FileReader(apiLogFilePath));
+        String logLine = bufferedReader.readLine();
+        Assert.assertNotNull(logLine);
+        while ((logLine = bufferedReader.readLine()) != null) {
+            Assert.assertTrue(logLine.contains("HTTP State Transition") ||
+                    logLine.contains("ROUND-TRIP LATENCY") || logLine.contains("Thread switch latency"));
+        }
+
+        sendGetRequest(offset, true);
+
+        while ((logLine = bufferedReader.readLine()) != null) {
+            Assert.assertTrue(logLine.contains("HTTP State Transition") ||
+                    logLine.contains("ROUND-TRIP LATENCY") || logLine.contains("Thread switch latency"));
+        }
+    }
+
+    /**
+     * Start a new server.
+     *
+     * @throws IOException
+     * @throws AutomationFrameworkException
+     */
+    private void startNewServer(boolean enableCorrelationLogs) throws IOException, AutomationFrameworkException {
+
+        HashMap<String, String> startupParameterMap = new HashMap<>();
+        startupParameterMap.put("-DportOffset", String.valueOf(offset));
+        startupParameterMap.put("-DenableCorrelationLogs", String.valueOf(enableCorrelationLogs));
+        startupParameterMap.put("startupScript", "micro-integrator");
+
+        server = new CarbonTestServerManager(context, System.getProperty("carbon.zip"), startupParameterMap);
+        server.startServer();
+        Awaitility.await().pollInterval(50, TimeUnit.MILLISECONDS).atMost(DEFAULT_TIMEOUT, TimeUnit.SECONDS).
+                until(isManagementApiAvailable());
+        carbonHome = server.getCarbonHome();
+        Assert.assertNotNull(carbonHome);
+    }
+
+    private void stopServer() throws AutomationFrameworkException {
+
+        server.stopServer();
+    }
+
+    private void sendGetRequest(int offset, boolean correlationConfigStatus) throws IOException {
```

Comment:
Something like `getCorrelationLogStateAndAssert` will be more meaningful. WDYT?

Reply:
Yes. Will change that